common/cards/alter-egos/single-use/ender-pearl.ts

```diff
@@ -48,25 +50,27 @@ class EnderPearlSingleUseCard extends SingleUseCard {
 				// We need to have no card there
 				if (pickResult.card) return 'FAILURE_INVALID_SLOT'
 
+				const activeRow = getActiveRowPos(player)
+				if (player.board.activeRow === null || !activeRow) return 'FAILURE_INVALID_DATA'
+
+				const logInfo = pickResult
```

Comment:
What type is logInfo, and what's it used for? Unclear to me

Reply:
Yeah.... this is the only part of the update I don't like.

I'm passing in pickInfo into applySingleUse when it exists so we can use it in the descriptions of cards. However, some cards make you pick a slot and we need to put the card you picked instead so I'm creating a new variable to override the card before passing it into applySingleUse. This is kind of a temporary solution, I don't mind removing it for now and coming back to those logs later.